pkg/apis/resource/validation/validation.go

```diff
@@ -693,3 +707,47 @@ func validateMap[K ~string, T any](m map[K]T, maxSize int, validateKey func(K, *
 	}
 	return allErrs
 }
+
+func validateDeviceStatus(device resource.AllocatedDeviceStatus, fldPath *field.Path, allocatedDevices sets.Set[structured.DeviceID]) field.ErrorList {
+	var allErrs field.ErrorList
+	deviceID := structured.DeviceID{Driver: device.Driver, Pool: device.Pool, Device: device.Device}
+	if !allocatedDevices.Has(deviceID) {
+		allErrs = append(allErrs, field.Invalid(fldPath, deviceID, "must be an allocated device in the claim"))
+	}
+	allErrs = append(allErrs, metav1validation.ValidateConditions(device.Conditions, fldPath.Child("conditions"))...)
+	if len(device.Data.Raw) > 0 { // Data is an optional field.
+		allErrs = append(allErrs, validateRawExtension(device.Data, fldPath.Child("data"))...)
+	}
+	allErrs = append(allErrs, validateNetworkDeviceData(device.NetworkData, fldPath.Child("networkData"))...)
+	return allErrs
+}
+
+// validateRawExtension validates RawExtension as in https://github.com/kubernetes/kubernetes/pull/125549/
+func validateRawExtension(rawExtension runtime.RawExtension, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	var v any
+	if len(rawExtension.Raw) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, ""))
+	} else if err := json.Unmarshal(rawExtension.Raw, &v); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, "<value omitted>", fmt.Sprintf("error parsing data: %v", err.Error())))
```

Comment:
I think @pohly has updated this function in https://github.com/kubernetes/kubernetes/pull/127511 - make sure you pick up his changes

Reply:
Yes, I included these changes and also from https://github.com/kubernetes/kubernetes/pull/128601